### ISSUE TITLE
Spring Boot starters should not bring log4j

### ISSUE THREAD

**Issue by snicoll** [SolaceProducts/solace-spring-boot#2]
A typical Spring Boot starter (i.e. something that relies on `spring-boot-starter`), as the starters of this project, relies on the logging starter which brings the logging infrastructure that Spring Boot uses by default. The current default is logback and users can switch to another logging system as [documented in the reference guide](https://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#howto-configure-log4j-for-logging).

Currently, the starter brings log4j2. That's breaking the current arrangement. I assume that Solace and the rest of the infrastructure uses slf4j and therefore log4j2 is not a strict requirement. Please consider excluding log4j in the starter (or elsewhere) so that the default arrangement described above is respected.

**Comment by dbruegger** [comment]
I agree with Stéphane. The `solace-java-spring-boot-starter` should **not** bring `log4j-core` as a dependency.

As the following Maven dependency tree listing shows, `log4j-core` comes as a transitive dependency through `solace-services-info`. As far as I can see, this is the only Solace dependency which logs directly to log4j instead of commons-logging.

In my opinion the best approach would be to use commons-logging in `solace-services-info`. If this is not possible, `log4j-core` should be excluded from `solace-java-spring-boot-starter` by a Maven dependency exclusion.

This issue is especially annoying with the current log4j zero-day exploit, see [Zero-Day Exploit Targeting Popular Java Library Log4j](https://www.govcert.ch/blog/zero-day-exploit-targeting-popular-java-library-log4j/).

```
[INFO] +- com.solace.spring.cloud:spring-cloud-starter-stream-solace:jar:3.2.0:compile
[INFO] |  \- com.solace.spring.cloud:spring-cloud-stream-binder-solace:jar:3.2.0:compile
[INFO] |     +- com.solace.spring.cloud:spring-cloud-stream-binder-solace-core:jar:3.2.0:compile
[INFO] |     |  \- org.springframework.boot:spring-boot-starter-cloud-connectors:jar:2.2.13.RELEASE:compile
[INFO] |     |     +- org.springframework.cloud:spring-cloud-spring-service-connector:jar:2.0.7.RELEASE:compile
[INFO] |     |     |  \- org.springframework.cloud:spring-cloud-connectors-core:jar:2.0.7.RELEASE:compile
[INFO] |     |     +- org.springframework.cloud:spring-cloud-cloudfoundry-connector:jar:2.0.7.RELEASE:compile
[INFO] |     |     +- org.springframework.cloud:spring-cloud-heroku-connector:jar:2.0.7.RELEASE:compile
[INFO] |     |     \- org.springframework.cloud:spring-cloud-localconfig-connector:jar:2.0.7.RELEASE:compile
[INFO] |     |        \- org.apache.commons:commons-lang3:jar:3.12.0:compile
[INFO] |     +- com.solacesystems:sol-jcsmp:jar:10.12.0:compile
[INFO] |     |  +- commons-logging:commons-logging:jar:1.2:compile
[INFO] |     |  +- commons-lang:commons-lang:jar:2.6:compile
[INFO] |     |  +- org.osgi:org.osgi.annotation:jar:6.0.0:compile
[INFO] |     |  \- org.apache.servicemix.bundles:org.apache.servicemix.bundles.jzlib:jar:1.0.7_2:compile
[INFO] |     +- com.solace.spring.boot:solace-java-spring-boot-starter:jar:4.1.0:compile
[INFO] |     |  \- com.solace.spring.boot:solace-java-spring-boot-autoconfigure:jar:4.1.0:compile
[INFO] |     |     +- com.solace.cloud.core:solace-services-info:jar:0.4.0:compile
[INFO] |     |     |  \- org.apache.logging.log4j:log4j-core:jar:2.14.1:compile
[INFO] |     |     \- com.solace.cloud.cloudfoundry:solace-java-cfenv:jar:1.1.0:compile
[INFO] |     |        +- io.pivotal.cfenv:java-cfenv-boot:jar:2.1.2.RELEASE:compile
[INFO] |     |        |  \- io.pivotal.cfenv:java-cfenv-jdbc:jar:2.1.2.RELEASE:compile
[INFO] |     |        |     \- io.pivotal.cfenv:java-cfenv:jar:2.1.2.RELEASE:compile
[INFO] |     |        \- com.google.code.gson:gson:jar:2.8.7:compile
[INFO] |     \- org.springframework.cloud:spring-cloud-stream:jar:3.1.3:compile
```

**Comment by GreenRover** [comment]
+1

**Comment by PhilippeKhalife** [comment]
We will action this shortly working on the source project for the dependency.

**Comment by Nephery** [comment]
Fixed with #11

**Comment by Jakingtosch** [comment]
@Nephery This ticket is not resolved by #11 
As you are fixing the `CVE-2021-45046`

In the meanwhile there is a new `CVE-2021-45105` where you have to upgrade the `log4j-core` dependency to `2.17`
This ticket here as described in the first and second comment of this thread addresses the underlying problem that you are using in one out of three solace owned libraries a different kind of log dependency with the `log4j-core` instead of `commons-logging`.

**Comment by Nephery** [comment]
@Jakingtosch #11 upgrades solace-services-info to `0.4.3` [which removes log4j entirely from solace-services-info](https://github.com/SolaceProducts/solace-services-info/commit/ea0a670c88fd78fdcaf141ceb7ec0cb9dbb50368).

So now, the only remaining usage of log4j comes from the Spring Boot Logging Starter and for that I recommend you check out https://spring.io/blog/2021/12/10/log4j2-vulnerability-and-spring-boot.

The log4j overrides you see in #11 only override's Spring's log4j in this project's tests and samples.
